src/braket/experimental/autoqasm/program/program.py

```diff
@@ -447,6 +448,74 @@ def push_oqpy_program(self, oqpy_program: oqpy.Program) -> None:
         finally:
             self._oqpy_program_stack.pop()
 
+    @contextlib.contextmanager
+    def if_block(self, condition: Any) -> None:
```

Comment:
Could these new functions be consolidated into one with a callback and args? They all do the same thing -- wrap control flow statements in a scope depth tracker. If you think it would make the code more confusing, then let's keep it as is.

Reply:
Thanks, this is a big improvement.